src/categorical_algebra/CSetDataStructures.jl

```diff
@@ -800,4 +802,47 @@ function deletesorted!(a::AbstractVector, x)
   found
 end
 
+""" More convenient syntax for declaring an ACSet
+
+Example:
+@acset Graph begin
+  V = 2
+  E = 2
+  src = [1,2]
+  tgt = [2,1]
+end
+"""
+
+macro acset(head, body)
+  expr = :(init_acset($(esc(head)), $(Expr(:quote, body))))
+  Expr(:call, esc(:eval), expr)
+end
+
+"""
+TODO: Well-formedness check
+TODO: Could also rely on a @generated function that took in a "flat" named tuple
+TODO: Alternative syntax for @acset input based on CSV
+TODO: Actual CSV input
+"""
+function init_acset(T::Type{<:ACSet{CD,AD,Ts}},body) where {CD <: CatDesc, AD <: AttrDesc{CD}, Ts <: Tuple}
+  body = strip_lines(body)
+  @assert body.head == :block
+  code = quote
+    acs = $(T)()
+  end
+  for elem in body.args
+    @assert elem.head == :(=)
```

Comment:
I do like to use `@match` for stuff like this:

```julia
lhs, rhs = @match elem begin
  Expr(:(=), lhs, rhs) => (lhs, rhs)
  _ => error("Every line of `@acset` must be an assignment")
end
```

In this case it's not any shorter but IMO it's easier to read.